docs/DeveloperGuide.md

```diff
@@ -292,6 +292,51 @@ Aspect: What fields should the `find` command search through?
   - Pros: Easy to implement, with minimal modification to existing tests.
   - Cons: Limited breadth of search, does not align with definition of unique application. 
 
+### Sort Feature
+
+#### Implementation
+
+The sort feature allows the user to sort the application list using company names, positions, application dates or interview dates. Each of these orders can also be reversed.
+
+The class diagram below shows the classes in the Logic component relevant for sorting:
+
+![Sort Class Diagram](images/SortClassDiagram.png)
+
+There is an abstract `SortCommand` class that inherits from the abstract `Command` class. Then, there is a concrete `SortCommand` subclass for each possible order of sort. Meanwhile, there is a single `SortCommandParser` class. When it parses the arguments supplied to a `sort` command, it decides which of the `SortCommand` subclasses to instantiate.
+
+The following sequence diagram shows the operation of the command `sort o/date r/`:
+![Sort Sequence Diagram](images/SortSequenceDiagram.png)
+
+When calling the `parse` method of the `SortCommandParser`, the argument provided for the `o/` prefix determines which subclass of `SortCommand` will get created. In the event that the prefix is not provided, a `SortByDateCommand` is returned by default.
+
+Later, when `LogicManager` `executes` the `SortCommand` created, the `SortCommand` will call one of the `sortApplicationList` methods provided by the `Model` interface for sorting the application list. Internally, the `Model` wraps its `ObservableList` of `Applications` inside a `SortedList`, so all it has to do is set an appropriate comparator on the `SortedList` to attain the desired sort order.
+
```

Comment:
I think can add limitation of sort feature (eg. the list is only able to sort with one criteria section)

Reply:
Thanks for the suggestion!